src/arena.rs

```diff
@@ -821,9 +821,9 @@ impl AllocSlab {
     }
 
     fn payload_offset<T>(&self) -> *mut T {
-        let mut ptr = (self as *const AllocSlab) as usize;
-        ptr += mem::size_of::<AllocSlab>();
-        ptr as *mut T
+        // This looks really scary, should this method be marked as unsafe?
+        // Also, this seems to cause UB.
```

Comment:
The UB here is probably related to provenance. Here it is returning the pointer to just after this object, and that is not a valid address for a pointer with the provenance of `&self` which is only allowed to access this object.